pkg/abstractions/function/function.go

```diff
@@ -123,8 +123,23 @@ func (fs *RunCFunctionService) FunctionInvoke(in *pb.FunctionInvokeRequest, stre
 }
 
 func (fs *RunCFunctionService) invoke(ctx context.Context, authInfo *auth.AuthInfo, stubId string, payload *types.TaskPayload) (types.TaskInterface, error) {
-	policy := types.DefaultTaskPolicy
-	policy.Expires = time.Now().Add(time.Duration(functionDefaultTaskExpiration) * time.Second)
+	stub, err := fs.backendRepo.GetStubByExternalId(ctx, stubId)
```

Comment:
Kind of a shame we have to query for this here on every function invocation -- aren't we also call this a level up?

```func (fs *RunCFunctionService) FunctionInvoke(in *pb.FunctionInvokeRequest, stream pb.FunctionService_FunctionInvokeServer) error {
	authInfo, _ := auth.AuthInfoFromContext(stream.Context())
	ctx := stream.Context()

	task, err := fs.invoke(ctx, authInfo, in.StubId, &types.TaskPayload{
		Args: []interface{}{in.Args},
	})
	if err != nil {
		return err
	}

	go func() {
		stub, err := fs.backendRepo.GetStubByExternalId(ctx, in.StubId)
		if err != nil {
			log.Printf("error getting stub: %v", err)
			return
		}

		go fs.eventRepo.PushRunStubEvent(authInfo.Workspace.ExternalId, &stub.Stub)
	}()

	return fs.stream(ctx, stream, authInfo, task)
}```